src/test/ui/async-await/argument-patterns.rs

```diff
@@ -0,0 +1,33 @@
+// edition:2018
+
+#![allow(unused_variables)]
+#![deny(unused_mut)]
+#![feature(async_await)]
+
+type A = Vec<u32>;
+
+async fn a(n: u32, mut vec: A) {
+    vec.push(n);
+}
+
+async fn b(n: u32, ref mut vec: A) {
+    vec.push(n);
+}
+
+async fn c(ref vec: A) {
+    vec.contains(&0);
+}
+
+async fn d((a, mut b): (A, A)) {
+    b.push(1);
+}
+
+async fn e((ref a, mut b): (A, A)) { //~ ERROR E0009
```

Comment:
Extract this bit out to a separate test so that this one can be `// compile-pass` so that we can be sure diagnostics aren't eaten.